### PR TITLE
Implement User Acivation

### PR DESCRIPTION
**Pull request by M-Jhay**
A draft PR to Implement User Activation

### CODE REVIEW

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 1]
```diff
@@ -1,10 +1,64 @@
...
 using Microsoft.AspNetCore.Mvc;
+using DwitTech.AccountService.Core.Interfaces;
+
 
 namespace DwitTech.AccountService.WebApi.Controllers
-{
+{ 
     public class UserController : BaseController
     {
+        public static User user = new User();
```
Comment (dabensymon) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: Why?

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+        private readonly IActivationService _activationService;
+
+        public UserController(IValidationService validationService, IActivationService activationService)
+        {
+            _validationService = validationService;
+            _activationService = activationService;
+        }
+
+
+        [HttpGet("/Account/Activation/{activationCode}")]
```
Comment (dabensymon) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: This should just be /Activation/{activationCode}. The Account in the path is inherited from the controller name.

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+        public IActionResult Get()
+        {
+            _activationService.GetActivationUrl();
```
Comment (dabensymon) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: Why this call here?

File: src/DwitTech.AccountService.Core/Interfaces/IValidationService.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Core.Interfaces
+{
+    public interface IValidationService
+    {
+        bool IsUniqueActivationCode();
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Interfaces/IValidationService.cs]: All these methods should not be here.

File: src/DwitTech.AccountService.Core/Services/ValidationService.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
+﻿using DwitTech.AccountService.Core.Interfaces;
+using DwitTech.AccountService.Core.Utilities;
+using DwitTech.AccountService.Data.Entities;
+using DwitTech.AccountService.Data.Context;
+using System.Net.Mail;
+
+namespace DwitTech.AccountService.Core.Services
+{
+    public class ValidationService : IValidationService
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/ValidationService.cs]: Do we really need this class? ActivationService should handle everything to do with UserActivation

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 1 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > Why?

Resolved

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 2 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > This should just be /Activation/{activationCode}. The Account in the path is inherited from the controller name.

Resolved

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 3 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > Why this call here?

Resolved

File: src/DwitTech.AccountService.Core/Interfaces/IValidationService.cs (hunk 4 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Interfaces/IValidationService.cs]: > All these methods should not be here.

Resolved

File: src/DwitTech.AccountService.Core/Services/ValidationService.cs (hunk 5 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/ValidationService.cs]: > Do we really need this class? ActivationService should handle everything to do with UserActivation

Resolved

File: src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs [hunk 6]
```diff
@@ -8,6 +9,7 @@ namespace DwitTech.AccountService.Core.Interfaces
 {
     public interface IActivationService
     {
-        bool SendActivationEmail(string fromEmail, string toEmail, string templateName, string RecipientName, string subject, string cc, string bcc);
+        
+        Task<bool> ActivateUser(string activationCode, string fromEmail, string toEmail, string templateName, string subject = "Account Details", string cc = "", string bcc = "");
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs]: There shouldn't be hardcoded values in your interface

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs [hunk 7]
```diff
@@ -60,12 +56,55 @@ public bool SendActivationEmail(string fromEmail, string toEmail, string templat
...
             templateText = templateText.Replace("{{activationUrl}}", activationUrl);
             string body = templateText;
             var response = SendMail(fromEmail, toEmail, subject, body, cc, bcc);
 
             return response;
         }
+
+        public bool SendWelcomeEmail(string fromEmail, string toEmail, string templateName, string subject = "Account Details", string cc = "", string bcc = "")
+        {
+            var user = new User();
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Core/Services/ActivationService.cs]: I don't see the benefit of this. You should remove it and accept a parameter of type User instead, or something along that line

File: src/DwitTech.AccountService.Core/Templates/WelcomeEmail.html [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+<body>
+    <h2>Dear {{Firstname}} {{Lastname}},</h2>
+
+    <p>
+        Welcome on board. Your account has been successfully set up and you can begin using it.
+
+        You will need the login details below to manage your dashboard
+        <br />
+        Email: {{Email}}
+        Password: {{Password}}
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Core/Templates/WelcomeEmail.html]: Will we have the password?
If the password will be hashed before storage, then I don't think this line is necessary because we can't send the user his hashed password, and even if we did have the original password, it would probably be a huge security risk sending it to the user via mail.

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs [hunk 9]
```diff
@@ -60,12 +56,55 @@ public bool SendActivationEmail(string fromEmail, string toEmail, string templat
...
+            string templateText = GetTemplate(templateName);
+            templateText = templateText.Replace("{{Firstname}}", user.FirstName);
+            templateText = templateText.Replace("{{Lastname}}", user.LastName);
+            templateText = templateText.Replace("{{Email}}", user.Email);
+            templateText = templateText.Replace("{{Password}}", user.Password);
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Core/Services/ActivationService.cs]: I don't think this line should be here. We don't have the password to send

File: src/DwitTech.AccountService.Data/Entities/UserStatus.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.AccountService.Data.Entities
+{
+    public enum UserStatus
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Data/Entities/UserStatus.cs]: If there's no db set for this, then I think it should be moved to Models.

File: src/DwitTech.AccountService.Data/repository/IUserRepository.cs [hunk 11]
```diff
@@ -9,7 +9,22 @@ namespace DwitTech.AccountService.Data.Repository
 {
     public interface IUserRepository
     {
-        
+        Task<ValidationCode> GetActivationDetail(string activationCode);
+        Task<bool> GetUserStatus(int id);
+        Task<bool> ValidateActivationCodeExpiry(string activationCode);
+        Task UpdateUserStatus(ValidationCode validationDetails);
+       
+        public class ValidationCode
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Data/repository/IUserRepository.cs]: You may need to sort this out with boss to know what will happen to it, cos I don't think it should be here. It'll probably cause a merge conflict with Maryann's migrations.

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs:1]: It's weird seeing method names that do not directly hint that they "belong" to the user entity, since this is supposed to be the User repository. I may be out of line, but it seems logical to add "User" somewhere in the method name. E.g GetUserActivationDetail, ValidateUserActivationCodeExpiry. Maybe seek a second opinion.

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs [hunk 12]
```diff
@@ -1,20 +1,63 @@
...
+                return true;
+            }
+            return false;
+        }
+
+        public async Task<bool> ValidateActivationCodeExpiry(string activationCode)
+        {
+            var validationCode = await GetActivationDetail(activationCode);
+
+            DateTime ExpiredTime = validationCode.CreatedTime.AddMinutes(10);
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.Data/repository/UserRepository.cs]: variable should begin with lowercase.

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 13]
```diff
@@ -1,10 +1,45 @@
...
+            _activationService = activationService;
+        }
+
+        
+        [HttpGet("/Activation/{activationCode}")]
+        public async Task<IActionResult> ActivateUser(string activationCode)
+        {
+            try
+            {
+                string fromEmail = "support@gmail";
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: These values should not be hardcoded

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 14]
```diff
@@ -1,10 +1,45 @@
...
+                string toEmail = "info@gmail.com";
+                string templateName = "WelcomeEmail";
+                string subject = "Account Details";
+                string cc = "";
+                string bcc = "";
+
+                var activationResult = await _activationService.ActivateUser(activationCode, fromEmail , toEmail, templateName, subject, cc, bcc);
+                return Ok(activationResult);
+            }
+            catch (Exception e)
```
Comment (giftedGeorge) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: I'm not sure about this, but I think you may need to handle exceptions differently here, cos It'll be very difficult to know what the error is if it returns 500 for all errors.

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 15]
```diff
@@ -1,20 +1,30 @@
...
+using System.Net.Sockets;
+using static DwitTech.AccountService.Data.Repository.IUserRepository;
 
 namespace DwitTech.AccountService.Core.Tests.Services
 {
     public class ActivationServiceTests
     {
+        private readonly IConfiguration _configuration;
+        private IActivationService actService;
+        private User mockUser;
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: This is not used. Please remove

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 16]
```diff
@@ -1,20 +1,30 @@
...
+        private ValidationCode mockValidationCode;
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: This too

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 17]
```diff
@@ -1,20 +1,30 @@
...
+        private Mock<IUserRepository> mockUserRepository;
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: And this

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 18]
```diff
@@ -1,20 +1,30 @@
...
+using Moq;
+using System.Net.Sockets;
+using static DwitTech.AccountService.Data.Repository.IUserRepository;
 
 namespace DwitTech.AccountService.Core.Tests.Services
 {
     public class ActivationServiceTests
     {
+        private readonly IConfiguration _configuration;
+        private IActivationService actService;
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: This is not used. Please remove

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 19]
```diff
@@ -1,20 +1,30 @@
...
-using System.Threading.Tasks;
+using Moq;
+using System.Net.Sockets;
+using static DwitTech.AccountService.Data.Repository.IUserRepository;
 
 namespace DwitTech.AccountService.Core.Tests.Services
 {
     public class ActivationServiceTests
     {
+        private readonly IConfiguration _configuration;
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: Since this is only used by your ActivateUser_Returns_BooleanResult test, I think it'll be better if you create it in your test instead of leaving it as a global variable.

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs [hunk 20]
```diff
@@ -24,13 +34,63 @@ public void SendActivationEmail_Returns_True(string fromMail, string toMail, str
...
                 .Build();
 
-            var actService = new ActivationService(configuration);
+            var options = new DbContextOptionsBuilder<AccountDbContext>()
+                .UseInMemoryDatabase(Guid.NewGuid().ToString())
+                .Options;
+
+            var mockDbContext = new Mock<AccountDbContext>(options);
+            IUserRepository repository = new UserRepository(mockDbContext.Object);
+            var iConfig = new Mock<IConfiguration>();
```
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: This line seems irrelevant since it is never passes into the ActivationService. A configuration variable exists in line 33, and is passed into the ActivationService in line 44, so iConfig is never used. Please explain its relevance to me if I'm wrong. Otherwise, you should remove it.

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs
Comment (giftedGeorge) [inline tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:1]: In my opinion, this test doesn't cover different scenarios and edge cases. Even if you may not cover all of them, you need to cover much more than just this.

**Review by giftedGeorge — COMMENTED** [review]
I think you did a good job, but there are some changes to be made. Aside from the comments I dropped, I also noticed there are no tests for the methods in your User repository class. Please confirm from boss if that is acceptable in your case. Otherwise, you'll have to add them

File: src/DwitTech.AccountService.Data/Entities/UserStatus.cs (hunk 10 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Data/Entities/UserStatus.cs]: > If there's no db set for this, then I think it should be moved to Models.

It is linked to the User entity, so the enum can be called from the User entity

File: src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs (hunk 6 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs]: > There shouldn't be hardcoded values in your interface

Resolved

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs (hunk 7 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/ActivationService.cs]: > I don't see the benefit of this. You should remove it and accept a parameter of type User instead, or something along that line

Resolved

File: src/DwitTech.AccountService.Core/Templates/WelcomeEmail.html (hunk 8 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Templates/WelcomeEmail.html]: > Will we have the password? If the password will be hashed before storage, then I don't think this line is necessary because we can't send the user his hashed password, and even if we did have the original password, it would probably be a huge security risk sending it to the user via mail.

Resolved

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs (hunk 9 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/ActivationService.cs]: > I don't think this line should be here. We don't have the password to send

Resolved

File: src/DwitTech.AccountService.Data/repository/IUserRepository.cs (hunk 11 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Data/repository/IUserRepository.cs]: > You may need to sort this out with boss to know what will happen to it, cos I don't think it should be here. It'll probably cause a merge conflict with Maryann's migrations.

Resolved

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs
Comment (M-Jhay) [reply src/DwitTech.AccountService.Data/repository/UserRepository.cs:1]: > It's weird seeing method names that do not directly hint that they "belong" to the user entity, since this is supposed to be the User repository. I may be out of line, but it seems logical to add "User" somewhere in the method name. E.g GetUserActivationDetail, ValidateUserActivationCodeExpiry. Maybe seek a second opinion.

Resolved

File: src/DwitTech.AccountService.Data/repository/UserRepository.cs (hunk 12 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Data/repository/UserRepository.cs]: > variable should begin with lowercase.

Resolved

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 13 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > These values should not be hardcoded

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 15 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > This is not used. Please remove

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 16 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > This too

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 17 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > And this

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 18 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > This is not used. Please remove

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 19 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > Since this is only used by your ActivateUser_Returns_BooleanResult test, I think it'll be better if you create it in your test instead of leaving it as a global variable.

Resolved

File: tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs (hunk 20 above)
Comment (M-Jhay) [reply tests/DwitTech.AccountService.Core.Tests/Services/ActivationServiceTests.cs]: > This line seems irrelevant since it is never passes into the ActivationService. A configuration variable exists in line 33, and is passed into the ActivationService in line 44, so iConfig is never used. Please explain its relevance to me if I'm wrong. Otherwise, you should remove it.

Resolved

File: tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs
Comment (M-Jhay) [reply tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs:1]: > In my opinion, this test doesn't cover different scenarios and edge cases. Even if you may not cover all of them, you need to cover much more than just this.

Well, I think the test is ok since I will be handling exceptions in the user controller

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 14 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > I'm not sure about this, but I think you may need to handle exceptions differently here, cos It'll be very difficult to know what the error is if it returns 500 for all errors.

Resolved

**Comment by M-Jhay** [comment]
> I think you did a good job, but there are some changes to be made. Aside from the comments I dropped, I also noticed there are no tests for the methods in your User repository class. Please confirm from boss if that is acceptable in your case. Otherwise, you'll have to add them

Resolved

File: src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs (hunk 6 above)
Comment (Maryann529) [reply src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs]: Why do you think there should not be hardcoded values in the interface?

**Review by johnny2565 — DISMISSED** [review]
The conflict resolutions and test fixes looks okay to me, boss

**Review by johnny2565 — DISMISSED** [review]
Updates where direct and simple, nice work

**Review by giftedGeorge — DISMISSED** [review]
Looks good to me

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs [hunk 21]
```diff
@@ -1,9 +1,31 @@
 ﻿using DwitTech.AccountService.Core.Interfaces;
+using System.Security.Cryptography;
+using System.Text;
 
 namespace DwitTech.AccountService.Core.Services
 {
     public class SecurityService : ISecurityService
     {
+
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/SecurityService.cs]: This shouldn't be here

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs [hunk 22]
```diff
@@ -1,10 +1,38 @@
...
+
+        public UserController(IActivationService activationService)
+        {
+           
+            _activationService = activationService;
+        }
+
+        
+        [HttpGet("/Activation/{activationCode}")]
+        public async Task<IActionResult> ActivateUser(string activationCode, User user, string fromEmail , string toEmail, string templateName, string subject, string cc, string bcc)
```
Comment (dabensymon) [inline src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: The only param available here is activationCode

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs [hunk 23]
```diff
@@ -60,12 +56,53 @@ public bool SendActivationEmail(string fromEmail, string toEmail, string templat
...
+
+        public async Task<bool> ActivateUser(string activationCode, User user, string fromEmail, string toEmail, string templateName, string subject = "Account Details", string cc = "", string bcc = "")
+        {
+            var activationResult = await _userRepository.GetUserActivationDetail(activationCode);
+            if (activationResult == null)
+            {
+                return false;
+            }
+        
+            var userStatus = await _userRepository.GetUserStatus(activationResult.UserId);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/ActivationService.cs]: This should be a call to GetUser

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs [hunk 24]
```diff
@@ -60,12 +56,53 @@ public bool SendActivationEmail(string fromEmail, string toEmail, string templat
...
+            string templateText = GetTemplate(templateName);
+            templateText = templateText.Replace("{{Firstname}}", user.Firstname);
+            templateText = templateText.Replace("{{Lastname}}", user.Lastname);
+            string body = templateText;
+            var response = SendMail(fromEmail, toEmail, subject, body, cc, bcc);
+
+            return response;
+        }
+
+        public async Task<bool> ActivateUser(string activationCode, User user, string fromEmail, string toEmail, string templateName, string subject = "Account Details", string cc = "", string bcc = "")
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Services/ActivationService.cs]: Method should take only activationCode

**Review by dabensymon — CHANGES_REQUESTED** [review]

File: src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs [hunk 25]
```diff
@@ -8,6 +9,7 @@ namespace DwitTech.AccountService.Core.Interfaces
 {
     public interface IActivationService
     {
-        bool SendActivationEmail(string fromEmail, string toEmail, string templateName, string RecipientName, string subject, string cc, string bcc);
```
Comment (dabensymon) [inline src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs:11]: why is this removed?

**Review by dabensymon — CHANGES_REQUESTED** [review]

File: src/DwitTech.AccountService.Core/Services/SecurityService.cs (hunk 21 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/SecurityService.cs]: > This shouldn't be here

Resolved

File: src/DwitTech.AccountService.WebApi/Controllers/UserController.cs (hunk 22 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.WebApi/Controllers/UserController.cs]: > The only param available here is activationCode

Resolved

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs (hunk 24 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/ActivationService.cs]: > Method should take only activationCode

Resolved

File: src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs (hunk 25 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Interfaces/IActivationService.cs:11]: > why is this removed?

Replaced

File: src/DwitTech.AccountService.Core/Services/ActivationService.cs (hunk 23 above)
Comment (M-Jhay) [reply src/DwitTech.AccountService.Core/Services/ActivationService.cs]: > This should be a call to GetUser

Resolved

**Review by johnny2565 — APPROVED** [review]
The changes looks okay to me boss

**Review by dabensymon — APPROVED** [review]
